### PR TITLE
Add `readonly ref readonly` example

### PR DESCRIPTION
**Pull request by Rageking8**
## Summary

The only 2 instances of `readonly ref readonly` being shown in examples are `ref` fields:
* https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-11.0/low-level-struct-improvements
* https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/ref-struct

This PR adds an example talking about `readonly ref readonly` methods, with the first `readonly` meaning readonly instance members instead of `readonly` ref fields like in the other 2 examples.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/csharp/language-reference/keywords/readonly.md](https://github.com/dotnet/docs/blob/5428171f426ee6007e9c4c778f4c4c000588dd7b/docs/csharp/language-reference/keywords/readonly.md) | [readonly (C# Reference)](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/readonly?branch=pr-en-us-37249) |

### CODE REVIEW

File: docs/csharp/language-reference/keywords/readonly.md [hunk 1]
```diff
@@ -73,6 +73,14 @@ The `readonly` modifier on a `ref return` indicates that the returned reference
 
 The type returned doesn't need to be a `readonly struct`. Any type that can be returned by `ref` can be returned by `ref readonly`.
 
+## Readonly ref readonly return example
+
+Ref readonly return can also be used in conjunction with readonly instance members:
```
Comment (BillWagner) [inline docs/csharp/language-reference/keywords/readonly.md:78]: Let's clarify that this is only valid on `struct` types:

```suggestion
Ref readonly return can also be used in conjunction with `readonly` instance members on `struct` types:
```

File: docs/csharp/language-reference/keywords/readonly.md [hunk 2]
```diff
@@ -73,6 +73,14 @@ The `readonly` modifier on a `ref return` indicates that the returned reference
...
+
+[!code-csharp[readonly ref readonly example](snippets/ReadonlyKeywordExamples.cs#ReadonlyRefReadonly)]
```
Comment (BillWagner) [inline docs/csharp/language-reference/keywords/readonly.md:80]: The preferred style uses the docs markdown extension for snippets:

```suggestion
:::code language="csharp" source="./snippets/ReadonlyKeywordExamples.cs" id="SnippetReadonlyRefReadonly":::
```

**Review by BillWagner — COMMENTED** [review]
Thanks for making this update @Rageking8 

I've got two small suggestions, and then it's ready to merge.

Thanks again

**Comment by Rageking8** [comment]
@BillWagner Applied all suggested changes. Thanks for the quick review!

**Review by BillWagner — APPROVED** [review]
Thanks again @Rageking8 

I'll :shipit: now.

**Pull request merged**